nobrainer/visualization/Guided-GradCAM.py

```diff
@@ -0,0 +1,31 @@
+import numpy as np
+from skimage.transform import resize
+import tensorflow as tf
+
+
+def Guided_GradCAM_3D(grad_model, ct_io, class_index):
+    # First outputs target convolution and output
+    input_ct_io = tf.expand_dims(ct_io, axis=-1)
+    input_ct_io = tf.expand_dims(input_ct_io, axis=0)
+    with tf.GradientTape() as tape:
+        conv_outputs, predictions = grad_model(input_ct_io)
+        loss = predictions[:, class_index]
+    # Extract filters and gradients
+    output = conv_outputs[0]
+    grads = tape.gradient(loss, conv_outputs)[0]
+    guided_grads = (
+        tf.cast(output > 0, "float32") * tf.cast(grads > 0, "float32") * grads
+    )
```

Comment:
i think this should be indented into the context:

```suggestion
        # Extract filters and gradients
        output = conv_outputs[0]
        grads = tape.gradient(loss, conv_outputs)[0]
        guided_grads = (
            tf.cast(output > 0, "float32") * tf.cast(grads > 0, "float32") * grads
        )
```